tick/optim/model/src/model.h

```diff
@@ -64,18 +64,28 @@ class Model {
     TICK_CLASS_DOES_NOT_IMPLEMENT(get_class_name());
   }
 
-  virtual double sdca_dual_min_i(ulong i,
-                                 const ArrayDouble &dual_vector,
+  virtual double sdca_dual_min_i(const ulong i,
+                                 const double dual_i,
                                  const ArrayDouble &primal_vector,
-                                 const ArrayDouble &previous_delta_dual,
+                                 const double previous_delta_dual_i,
                                  double l_l2sq) {
     TICK_CLASS_DOES_NOT_IMPLEMENT(get_class_name());
   }
 
+  virtual SArrayULongPtr get_sdca_index_map(){
+    return nullptr;
```

Comment:
No, it should not raise an error, most models do not need an sdca index map as it would simply be identity (index_map[i] = i). But It should be more explicit that nullptr means identity.